### DISCUSSION TITLE
Select base model at post api, based on value of path parameter

### DISCUSSION THREAD

**Discussion by shrikantnarayankar15** [fastapi/fastapi#6642] [Questions] [answered]
Hi,
    I want help on the below issue,
I have 3 basemodel schema A, B, C. 

Suppose, 
@app.post('/{type}/')
async def createData(type:str, item: A)
       return item
In the above, "item : A", this I want to change at runtime. Based on the path parameter value in this case "type".

Python :3.9

**Comment by dstlny** [comment, marked as answer]
> Hi,
> I want help on the below issue,
> I have 3 basemodel schema A, B, C.
> 
> Suppose,
> @app.post('/{type}/')
> async def createData(type:str, item: A)
> return item
> In the above, "item : A", this I want to change at runtime. Based on the path parameter value in this case "type".
> 
> Python :3.9

Do something like:

```python
@app.post('/{type}/')
async def createData(
     type: str,
     item: Union[A, B, C] = Depends(get_item) ## or just have a common base class used in all the items, and use that in the annotation
):
     # do something


def get_item(type: str):
      # do something
      return item
```

**Reply by YuriiMotov** [reply]
This way you will not be able to specify in openapi the relations between the value of `type` and the expected schema of `item` (you can do it in description, but it's not good if you use code generators).

If it's not a problem, the example described above is fine. You can add validation logic to `get_item` to ensure `item` is of correct type. 

If you need more strict openapi schema, create separate endpoints for each type. You can move common logic of endpoint function into separate function to reduce code duplication

**Comment by shrikantnarayankar15** [comment]
Thanks
